pkg/controller/infinispan/infinispan_controller.go

```diff
@@ -980,8 +1006,43 @@ func (r *ReconcileInfinispan) statefulSetForInfinispan(m *infinispanv1.Infinispa
 	if _, err := applyExternalArtifactsDownload(m, &dep.Spec.Template.Spec); err != nil {
 		return nil, err
 	}
-
 	applyExternalDependenciesVolume(m, &dep.Spec.Template.Spec)
+
+	// ISPN-12667 ISPN-13121 If upgrading the cluster ensure that cache .state required properties are added
+	if m.IsConditionTrue(infinispanv1.ConditionGracefulShutdown) {
```

Comment:
@ryanemerson do we need update init container in the main branch? if we will not skip previous operator version it will update to the correct state. Or I'm wrong?

Reply:
That's true. It also shouldn't be necessary to have the check for the 11223 port if we make this assumption. I'll update to remove these two commits.